app/routes/goal_routes.py

```diff
@@ -1 +1,87 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response, Response
+from .route_utilities import validate_model
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal)
+    goals = db.session.scalars(query)
+    goals_response = [goal.to_dict() for goal in goals]
+    return goals_response
+
+
+@bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal": goal.to_dict()}
+
+
+@bp.put("/<goal_id>")
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+
+    db.session.add(goal)
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
```

Comment:
In `goal_routes.py` I updated lines 33 and 59. The rest of the implementation is unchanged.